ui/app/templates/components/date-dropdown.hbs

```diff
@@ -4,7 +4,7 @@
 ~}}
 
 <Hds::SegmentedGroup ...attributes as |S|>
-  <S.Dropdown @height="200px" as |dd|>
+  <S.Dropdown @listPosition="bottom-left" @height="200px" as |dd|>
```

Comment:
Fixes this issue at smaller screens:
<img width="845" alt="start-edit-before" src="https://github.com/hashicorp/vault/assets/82459713/c5897e33-40bc-4822-9b94-2f3977efa4b5">

After: 
<img width="845" alt="start-edit-after" src="https://github.com/hashicorp/vault/assets/82459713/c93769c6-2078-4c70-97e2-cea9fe4969ee">